Rendera/include/cube.hpp

```diff
@@ -0,0 +1,155 @@
+#ifndef CUBE_HPP
+#define CUBE_HPP
+
+#include <limits>
+#include "vec3.hpp"
+#include "ray.hpp"
+#include "material.hpp"
+
+
+vec3 max(vec3 a, vec3 b, ray r)
+{
+    if (dot((a - r.get_origin()),r.get_direction()) > dot((b- r.get_origin()),r.get_direction()))
+    {
+        return a;
+    }
+    else
+    {
+        return b;
+    }
+}
+
+vec3 min(vec3 a, vec3 b, ray r)
+{
+
+    if (dot((a - r.get_origin()),r.get_direction()) < dot((b- r.get_origin()),r.get_direction()))
+    {
+        return a;
+    }
+    else
+    {
+        return b;
+    }
+}
+
+class Cube
+{
+private:
+    vec3 m_position;
+    vec3 m_a, m_b, m_c;
+    Material m_material;
+
+public:
+    Cube(vec3 position, vec3 a, vec3 b, vec3 c, Material mat) : m_position{position}, m_a{a}, m_b{b}, m_c{c}, m_material{mat} {};
+
+    ray normal(const vec3 &point)
+    {
+        vec3 p_pos, np_pos;
+        p_pos = point - m_position;
+        np_pos = point - m_a - m_b - m_c - m_position;
+        vec3 normal;
+        if (dot(p_pos, cross(m_a, m_b)) < 1e5)
+        {
+            normal = cross(m_a, m_b);
+            if (dot(m_c, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(p_pos, cross(m_a, m_c)) < 1e5)
+        {
+            normal = cross(m_a, m_c);
+            if (dot(m_b, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(p_pos, cross(m_b, m_c)) < 1e5)
+        {
+            normal = cross(m_b, m_c);
+            if (dot(m_a, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(np_pos, cross(m_a, m_b)) < 1e5)
+        {
+            normal = cross(m_a, m_b);
+            if (dot(-m_c, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(np_pos, cross(m_a, m_c)) < 1e5)
+        {
+            normal = cross(m_a, m_c);
+            if (dot(-m_b, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else if (dot(np_pos, cross(m_b, m_c)) < 1e5)
+        {
+            normal = cross(m_b, m_c);
+            if (dot(-m_a, normal) < 0)
+            {
+                normal = -normal;
+            }
+        }
+        else
+        {
+            normal = vec3(0, 0, 0);
+        }
+        return ray(point, normal);
+    }
+
+
+    vec3 plane_intersect(const vec3 &n, const vec3 &p, const ray &r)
+    {
+        float num = dot(n, p - r.get_origin());
+        if (abs(dot(r.get_direction(), n)) > 1e-5)
```

Comment:
Similarly, a comment here, explaining what this function does, would be nice too.